File: Part8/Fortran/FunctionConcepts/FunctionConcepts_11.f90
! Computing factorial with loop
program main
    integer :: Factorial_Result
    
    Factorial_Result = factorial ( 10 )
    print *, "Factorial of 10 is:", Factorial_Result
    
contains
    function factorial ( Factorial_Number_Par ) result ( Factorial_Result )
        integer, intent(in) :: Factorial_Number_Par
        integer :: Factorial_Result
        integer :: Total_Factorial = 1
        integer :: Factorial_Number
        
        do Factorial_Number = 1, Factorial_Number_Par
            Total_Factorial = Total_Factorial * Factorial_Number
        end do
        
        Factorial_Result = Total_Factorial
    end function factorial
end program main

File: Part8/Go/FunctionConcepts/FunctionConcepts_11.go
// Recursive function call to compute factorial
package main

import "fmt"

func factorial ( FactorialNumber_Par int ) int {
    if FactorialNumber_Par <= 1 {
        return 1
    }
    
    return FactorialNumber_Par * factorial ( FactorialNumber_Par - 1 )
}

func main() {
    factorialResult := factorial ( 10 )
    fmt.Printf ( "Factorial of 10 is: %d\n", factorialResult )
}
